Components/TriggerService.php

```diff
@@ -0,0 +1,271 @@
+        $this->createArticlePricesTrigger();
+        $this->createArticleTranslationsTrigger();
+        $this->createSupplierTrigger();
+        $this->createTaxTrigger();
+        $this->createCategoriesTrigger();
+    }
+    public function deactivateTriggers()
+    {
+        $this->connection->executeQuery('
+            DROP TRIGGER IF EXISTS connect_article_trigger;
+            DROP TRIGGER IF EXISTS connect_article_details_update_trigger;
+            DROP TRIGGER IF EXISTS connect_article_details_insert_trigger;
+            DROP TRIGGER IF EXISTS connect_article_details_delete_trigger;
+            DROP TRIGGER IF EXISTS connect_article_attributes_update_trigger;
+            DROP TRIGGER IF EXISTS connect_article_attributes_insert_trigger;
+            DROP TRIGGER IF EXISTS connect_article_attributes_delete_trigger;
+            DROP TRIGGER IF EXISTS connect_article_images_update_trigger;
+            DROP TRIGGER IF EXISTS connect_article_images_insert_trigger;
+            DROP TRIGGER IF EXISTS connect_article_images_delete_trigger;
+            DROP TRIGGER IF EXISTS connect_article_categories_update_trigger;
+            DROP TRIGGER IF EXISTS connect_article_categories_insert_trigger;
+            DROP TRIGGER IF EXISTS connect_article_categories_delete_trigger;
+            DROP TRIGGER IF EXISTS connect_article_prices_update_trigger;      
+            DROP TRIGGER IF EXISTS connect_article_prices_insert_trigger;
+            DROP TRIGGER IF EXISTS connect_article_prices_delete_trigger;
+            DROP TRIGGER IF EXISTS connect_article_translations_update_trigger;
+            DROP TRIGGER IF EXISTS connect_article_translations_insert_trigger;
+            DROP TRIGGER IF EXISTS connect_article_translations_delete_trigger;
+            DROP TRIGGER IF EXISTS connect_supplier_trigger;
+            DROP TRIGGER IF EXISTS connect_tax_trigger;  
+            DROP TRIGGER IF EXISTS connect_categories_trigger;  
+        ');
+    }
+
+
+    private function createArticleTrigger()
+    {
+        $this->connection->executeQuery('
+            CREATE TRIGGER connect_article_trigger
+            AFTER UPDATE
+            ON s_articles
+            FOR EACH ROW
+                UPDATE s_plugin_connect_items SET cron_update = 1 WHERE article_id = NEW.id
+        ');
+    }
+
+    private function createArticleDetailsTrigger()
+    {
+        $this->connection->executeQuery('
+            CREATE TRIGGER connect_article_details_update_trigger
+            AFTER UPDATE
+            ON s_articles_details
+            FOR EACH ROW
+                UPDATE s_plugin_connect_items SET cron_update = 1 WHERE article_id = NEW.articleID
+        ');
+        $this->connection->executeQuery('
+            CREATE TRIGGER connect_article_details_insert_trigger
+            AFTER INSERT
+            ON s_articles_details
+            FOR EACH ROW
+                UPDATE s_plugin_connect_items SET cron_update = 1 WHERE article_id = NEW.articleID
```

Comment:
that is a really good point
before it was handled in our lifecycle subscriber
if added an check in the CronJob so that just exported products will be processed
so exported = 1 and cron_update = 1 has to be set to be updated
i think this will work in every case because the first export for each product is triggered manually over the export view